### PR TITLE
Fix Support withFreeTier, withStandardTier and withPremiumTier for aks (issue#38881)

### PR DESCRIPTION
**Pull request by v-hongli1**
relate to [issue#38881](https://github.com/Azure/azure-sdk-for-java/issues/38881)

# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/resourcemanager/azure-resourcemanager-containerservice/src/test/java/com/azure/resourcemanager/containerservice/KubernetesClustersTests.java [hunk 1]
```diff
@@ -530,4 +531,343 @@ public void testFipsEnabled() {
...
+            .withVirtualMachineSize(ContainerServiceVMSizeTypes.STANDARD_D2_V2)
+            .withAgentPoolVirtualMachineCount(1)
+            .withAgentPoolType(AgentPoolType.VIRTUAL_MACHINE_SCALE_SETS)
+            .withAgentPoolMode(AgentPoolMode.SYSTEM)
+            .attach()
+            .withDnsPrefix(generateRandomResourceName("dns", 15))
+            .withFreeTier()
+            .create();
+        kubernetesCluster.refresh();
+        Assertions.assertEquals(ManagedClusterSkuTier.FREE, kubernetesCluster.innerModel().sku().tier());
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-containerservice/src/test/java/com/azure/resourcemanager/containerservice/KubernetesClustersTests.java:583]: We need a getter for tier as well. Maybe just `sku()`?

File: sdk/resourcemanager/azure-resourcemanager-containerservice/src/test/java/com/azure/resourcemanager/containerservice/KubernetesClustersTests.java [hunk 2]
```diff
@@ -530,4 +531,343 @@ public void testFipsEnabled() {
...
+        kubernetesCluster.refresh();
+        Assertions.assertEquals("1.27", kubernetesCluster.version());
+
+        kubernetesCluster.update().withVersion("1.28.5").apply();
+        kubernetesCluster.refresh();
+        Assertions.assertEquals("1.28.5", kubernetesCluster.version());
+    }
+
+    @Test
+    public void testCreateKubernetesClusterWithFreePricingTier() {
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-containerservice/src/test/java/com/azure/resourcemanager/containerservice/KubernetesClustersTests.java:563]: Do we need the "create only" tests? Aren't they covered in the following "from to" tests?

File: sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/models/KubernetesCluster.java [hunk 3]
```diff
@@ -188,6 +189,33 @@ interface Blank extends DefinitionWithRegion<WithGroup> {
         interface WithGroup extends GroupableResource.DefinitionStages.WithGroup<WithVersion> {
         }
 
+        /**
+         * The stage of the Kubernetes cluster definition allowing to specify AKS pricing tier.
+         */
+        interface WithPricingTier {
+
+            /**
+             * Specifies the pricing tier  the Kubernetes cluster is free.
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/models/KubernetesCluster.java:198]: nit, double spaces between `tier` and `the`

Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/models/KubernetesCluster.java]: Fixed in the new version.

File: sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/models/KubernetesCluster.java [hunk 4]
```diff
@@ -188,6 +192,33 @@ interface Blank extends DefinitionWithRegion<WithGroup> {
...
+         * The stage of the Kubernetes cluster definition allowing to specify a tier for the managed cluster SKU.
+         */
+        interface WithPricingTier {
+
+            /**
+             * Specifies the tier of the managed cluster SKU is free.
+             *
+             * @return the next stage of the definition
+             */
+            WithCreate withFreeTier();
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/models/KubernetesCluster.java:205]: Seems we usually name it `withFreePricingTier` or `withFreeSku`

File: sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/implementation/KubernetesClusterImpl.java [hunk 5]
```diff
@@ -436,6 +445,27 @@ private void clearKubeConfig() {
...
+    public KubernetesClusterImpl withStandardTier() {
+        this.innerModel().withSku(new ManagedClusterSku().withTier(ManagedClusterSkuTier.STANDARD).withName(ManagedClusterSkuName.BASE));
+        this.innerModel().withSupportPlan(KubernetesSupportPlan.KUBERNETES_OFFICIAL);
+        return this;
+    }
+
+    @Override
+    public KubernetesClusterImpl withPremiumTier() {
+        this.innerModel().withSku(new ManagedClusterSku().withTier(ManagedClusterSkuTier.PREMIUM).withName(ManagedClusterSkuName.BASE));
+        this.innerModel().withSupportPlan(KubernetesSupportPlan.AKSLONG_TERM_SUPPORT);
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/implementation/KubernetesClusterImpl.java:465]: Any docuemnt/detail on the SupportPlan on different SKU?

File: sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/models/KubernetesCluster.java [hunk 6]
```diff
@@ -727,5 +761,53 @@ interface WithLocalAccounts {
...
+             * @return the next stage
+             */
+            Update withVersion(String kubernetesVersion);
+
+            /**
+             * Uses the default version for the Kubernetes cluster.
+             *
+             * @return the next stage
+             */
+            Update withDefaultVersion();
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/models/KubernetesCluster.java:810]: Could cluster be updated to default version?

File: sdk/resourcemanager/azure-resourcemanager-containerservice/src/test/java/com/azure/resourcemanager/containerservice/KubernetesClustersTests.java [hunk 7]
```diff
@@ -530,4 +531,271 @@ public void testFipsEnabled() {
...
+        kubernetesCluster.refresh();
+        Assertions.assertEquals(ManagedClusterSkuTier.STANDARD, kubernetesCluster.innerModel().sku().tier());
+
+        kubernetesCluster.update().withFreeTier().apply();
+        kubernetesCluster.refresh();
+        Assertions.assertEquals(ManagedClusterSkuTier.FREE, kubernetesCluster.innerModel().sku().tier());
+    }
+
+    @Test
+    public void  testUpdatePricingTierFromFreeToPremium() {
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-containerservice/src/test/java/com/azure/resourcemanager/containerservice/KubernetesClustersTests.java:691]: We probably don't need that many tests of xxTier to yyTier.

File: sdk/resourcemanager/azure-resourcemanager-containerservice/src/test/java/com/azure/resourcemanager/containerservice/KubernetesClustersTests.java [hunk 8]
```diff
@@ -530,4 +531,271 @@ public void testFipsEnabled() {
...
+            .withVirtualMachineSize(ContainerServiceVMSizeTypes.STANDARD_D2_V2)
+            .withAgentPoolVirtualMachineCount(1)
+            .withAgentPoolType(AgentPoolType.VIRTUAL_MACHINE_SCALE_SETS)
+            .withAgentPoolMode(AgentPoolMode.SYSTEM)
+            .attach()
+            .withDnsPrefix(generateRandomResourceName("dns", 15))
+            .create();
+        kubernetesCluster.refresh();
+        kubernetesCluster.update().withFreeTier().apply();
+        Assertions.assertEquals(ManagedClusterSkuTier.FREE, kubernetesCluster.innerModel().sku().tier());
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-containerservice/src/test/java/com/azure/resourcemanager/containerservice/KubernetesClustersTests.java:583]: `kubernetesCluster.sku()`

File: sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/models/KubernetesCluster.java (hunk 6 above)
Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/models/KubernetesCluster.java]: After testing, it cannot be updated from the specified version to the default version, so `withDefaultVersion` is removed from `UpdateStages`.

File: sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/models/KubernetesCluster.java (hunk 4 above)
Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/models/KubernetesCluster.java]: The name changed to `withFreeSku`,  the `withStandardTier` and `withPremiuTier` make the same fix.

File: sdk/resourcemanager/azure-resourcemanager-containerservice/src/test/java/com/azure/resourcemanager/containerservice/KubernetesClustersTests.java (hunk 8 above)
Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-containerservice/src/test/java/com/azure/resourcemanager/containerservice/KubernetesClustersTests.java]: The `kubernetesCluster.innerModel().sku().tier()` has been changed to `kubernetesCluster.sku()`.

File: sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/implementation/KubernetesClusterImpl.java [hunk 9]
```diff
@@ -436,6 +445,31 @@ private void clearKubeConfig() {
         this.formatUserKubeConfigsMap.clear();
     }
 
+    @Override
+    public KubernetesClusterImpl withFreeSku() {
+        if (!this.isInCreateMode() && ManagedClusterSkuTier.PREMIUM.equals(this.innerModel().sku().tier())) {
+            this.innerModel().withSupportPlan(KubernetesSupportPlan.KUBERNETES_OFFICIAL);
+        }
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/implementation/KubernetesClusterImpl.java:452]: According to the docs, FREE and STANDARD can set the plan to LTS, but not charged until upgrade to PREMIUM:
https://learn.microsoft.com/en-us/azure/aks/long-term-support#enable-long-term-support

If that's true, we probably need a separate API for plan.

File: sdk/resourcemanager/azure-resourcemanager-containerservice/src/test/java/com/azure/resourcemanager/containerservice/KubernetesClustersTests.java [hunk 10]
```diff
@@ -530,4 +531,271 @@ public void testFipsEnabled() {
...
+        kubernetesCluster.refresh();
+        Assertions.assertEquals("1.27", kubernetesCluster.version());
+
+        kubernetesCluster.update().withVersion("1.28.5").apply();
+        kubernetesCluster.refresh();
+        Assertions.assertEquals("1.28.5", kubernetesCluster.version());
+    }
+
+    @Test
+    public void testUpdatePricingTierFromDefaultToFree() {
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-containerservice/src/test/java/com/azure/resourcemanager/containerservice/KubernetesClustersTests.java:563]: Can we have one test cover all the cases(e.g. create FREE, update to STANDARD, update to PREMIUM, etc.)?

**Pull request closed without merging**